packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,678 @@
+   *
+   * @returns 'spark'
+   */
+  static readonly SPARK = new Classification('spark');
+
+  /**
+   * Sets values in the spark-defaults.conf file.
+   *
+   * For more info:
+   * @see https://spark.apache.org/docs/latest/configuration.html
+   *
+   * @returns 'spark-defaults'
+   */
+  static readonly SPARK_DEFAULTS = new Classification('spark-defaults');
+
+  /**
+   * Sets values in the spark-env.sh file.
+   *
+   * For more info:
+   * @see https://spark.apache.org/docs/latest/configuration.html#environment-variables
+   *
+   * @returns 'spark-env'
+   */
+  static readonly SPARK_ENV = new Classification('spark-env');
+
+  /**
+   * Sets values in the hive-site.xml for Spark.
+   *
+   * @returns 'spark-hive-site'
+   */
+  static readonly SPARK_HIVE_SITE = new Classification('spark-hive-site');
+
+  /**
+   * Sets values in the log4j.properties file.
+   *
+   * For more settings and info:
+   * @see https://github.com/apache/spark/blob/master/conf/log4j.properties.template
+   *
+   * @returns 'spark-log4j'
+   */
+  static readonly SPARK_LOG4J = new Classification('spark-log4j');
+
+  /**
+   * Sets values in the metrics.properties file.
+   *
+   * For more settings and info:
+   * @see https://github.com/apache/spark/blob/master/conf/metrics.properties.template
+   *
+   * @returns 'spark-metrics'
+   */
+  static readonly SPARK_METRICS = new Classification('spark-metrics');
+
+  /**
+   * Creates a new Classification, can be extended to support a classification
+   *
+   * @param classificationStatement A literal string in case a new EMR classification is released, if not already defined.
+   */
+  constructor(public readonly classificationStatement: string) { }
+}
+
+/**
+ * A configuration specification to be used when provisioning virtual clusters,
+ * which can include configurations for applications and software bundled with Amazon EMR on EKS.
+ *
+ * A configuration consists of a classification, properties, and optional nested configurations.
+ * A classification refers to an application-specific configuration file.
+ * Properties are the settings you want to change in that file.
+ * @see https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-configure-apps.html
+ */
+export interface ApplicationConfiguration {
+
+  /**
+   * The classification within a configuration.
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 1024.
+   */
+  readonly classification: Classification;
+
+  /**
+   * A list of additional configurations to apply within a configuration object.
+   *
+   * Array Members: Maximum number of 100 items.
+   *
+   * @default - No other configurations
+   */
+  readonly nestedConfig?: ApplicationConfiguration[];
+
+  /**
+   * A set of properties specified within a configuration classification.
+   *
+   * Map Entries: Maximum number of 100 items.
+   *
+   * @default - No properties
+   */
+  readonly properties?: { [key: string]: string };
```

Comment:
does it make sense for an app config to be supplied with no properties? maybe this should be required

Reply:
Yes, you can just use the classification, requiring this wouldn't make sense.

Reply:
but what would an app config with just a classification and no properties actually do?

Reply:
The properties are the settings that you want to change within the application specific configuration file that's referred to by the classification.

So if customers set the classification to ``spark defaults`` for example, customers don't have to modify anything to that spark defaults app config file. IF they do, then they can use the ``properties`` parameter to do so.

Reply:
If providing an app config with just a classification and no properties is a no-op, then we should not allow it. Optional truly means optional; if an optional property is not provided then it's because the rest of the properties have some sort of meaning without it. Make this required

Reply:
It's not a no-op. [spark defaults](https://spark.apache.org/docs/latest/configuration.html#viewing-spark-properties)

Reply:
I understand there are defaults. If a user wants the defaults then they should not provide an app config. Having an empty app config helps nobody